pkg/attest/platform_cert_fetcher.go

```diff
@@ -166,33 +173,36 @@ func fetchWithRetry(requestURL string, baseSec int, maxRetries int) ([]byte, err
 		}
 		res, err := http.Get(requestURL)
 		if err != nil {
+			logrus.Debugf("fetch on retry %d: http.Get failed: %s", retryCount, err)
 			retryCount++
 			continue
 		}
 		if 200 <= res.StatusCode && res.StatusCode < 300 {
 			// Got successful status code 2xx
 			defer res.Body.Close()
-			resBody, err := ioutil.ReadAll(res.Body)
+			resBody, err := io.ReadAll(res.Body)
 			if err != nil {
+				logrus.Debugf("fetch on retry %d: http.Get failed: %s", retryCount, err)
 				retryCount++
 				continue
 			}
 			return resBody, nil
 		} else if res.StatusCode == 408 || res.StatusCode == 429 || 500 <= res.StatusCode {
 			// Got status code that is worth to retry
+			logrus.Debugf("fetch on retry %d: http.Get failed with a status code worth a retry", retryCount)
 			retryCount++
 			continue
 		} else {
 			// Got status code that is not worth to retry
 			defer res.Body.Close()
-			resBody, err := ioutil.ReadAll(res.Body)
+			resBody, err := io.ReadAll(res.Body)
 			if err != nil {
-				return nil, fmt.Errorf("got error while handling non successful response with status code %d: %s", res.StatusCode, err)
+				return nil, errors.Errorf("got error while handling non successful response with status code %d: %s", res.StatusCode, err)

```

Comment:
Error responses from servers will commonly include helpful information in the resonse body.  Even if there is an error in reading the response body, you may have already read some bytes that could be useful.